### DISCUSSION TITLE
import upip error

### DISCUSSION THREAD

**Discussion by travelo94** [micropython/micropython#12755] [General]
Hi , 
Im trying to connect Rpi W to the wifi and mqtt broker , 
I was following this blog 
https://www.hivemq.com/article/iot-reading-sensor-data-raspberry-pi-pico-w-micropython-mqtt-node-red/
But when i try to import upip to then install umqtt simple i get error
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/lib/upip.py", line 13, in <module>
ImportError: no module named 'uzlib'
```
How can i upload the missing module?

**Comment by andrewleech** [comment]
`upip` tool was replaced some time ago with `mip`
https://docs.micropython.org/en/latest/reference/packages.html
https://www.tomshardware.com/how-to/raspberry-pi-pico-micropython-mip

Maybe check you've got the latest release firmware installed and use the mip guidance to get the mqtt library installed, hopefully the rest of the tutorial of still valid

**Reply by travelo94** [reply]
Thanks , i managed to install the library manually
